api/enterprise-api-ref/build-api.markdown

```diff
@@ -833,3 +833,164 @@ HTTP 200 OK
 | 200 Ok | Successful response  |
 | 404 Not found | Module not found  |
 | 500 Internal server error | Internal server error |
+
+
+## Get CFEngine Build module input data
+
+**URI:** https://hub.cfengine.com/api/build/projects/:id/modules/:name/input
+
+**Method:** GET
+
+**Parameters:**
+
+* **id** *(integer)*
+  Project's ID. Required.
+* **name** *(string)*
+  Module name. Required.
+
+**Example request (curl):**
+
+```console
+curl --user <username>:<password> \
+  -X GET \
+  https://hub.cfengine.com/api/build/projects/1/modules/delete-files/input
+```
+
+**Successful response example:**
+
+```
+HTTP 200 OK
+{
+    "status": "ok",
+    "input_spec": [
+        {
+            "type": "list",
+            "label": "Files",
+            "while": "Specify another file you want deleted on your hosts?",
+            "bundle": "delete_files",
+            "subtype": [
+                {
+                    "key": "path",
+                    "type": "string",
+                    "label": "Path",
+                    "question": "Path to file"
+                },
+                {
+                    "key": "why",
+                    "type": "string",
+                    "label": "Why",
+                    "default": "Unknown",
+                    "question": "Why should this file be deleted?"
+                }
+            ],
+            "response": [
+                {
+                    "path": "/tmp/test",
+                    "why": "no tests, please"
+                }
+            ],
+            "variable": "files",
+            "namespace": "delete_files"
+        }
+    ]
+}
+```
+
+**Output:**
+
+* **input_spec** *(JSON array of objects)*
+  Input specification represented as an JSON array of objects. 
+  Each object specifies type, label, bundle, while, variable, namespace, 
+  subtype (in case of list type) and response (when a module has input data) properties.
```

Comment:
Not every object will have `while` and `subtype`, simple string values don't have/need them. I would write something like *Each object specifies one input entry for the module. To discover more about the fields...*

Reply:
done